src/rootPages/Designer/ui_work_datacollection_workspace_properties.js

```diff
@@ -0,0 +1,1036 @@
+                  id: "_FirstRecord",
+                  value: L("[First Record]"),
+               },
+               {
+                  id: "_FirstRecordDefault",
+                  value: L("[Default to First Record]"),
+               }
+            );
+         }
+
+         dataItems.unshift({
+            id: "",
+            value: L("Select fix cursor"),
+         });
+
+         fixSelect = this.CurrentDatacollection.settings.fixSelect || "";
+
+         $$(ids.fixSelect).define("options", dataItems);
+         $$(ids.fixSelect).define("value", fixSelect);
+         $$(ids.fixSelect).refresh();
+      }
+
+      initPopupEditors() {
+         this.FilterComponent.init({
+            // when we make a change in the popups we want to make sure we save the new workspace to the properties to do so just fire an onChange event
+            onChange: this.onFilterChange,
+         });
+
+         this.PopupSortFieldComponent.init(this.AB);
+      }
+
+      selectSource(datasourceID, oldId) {
+         const ids = this.ids;
+         const selectedDatasource = $$(ids.dataSource)
+            .getList()
+            .getItem(datasourceID);
+
+         if (selectedDatasource && selectedDatasource.disabled) {
+            // prevents re-calling onChange from itself
+            $$(ids.dataSource).blockEvent();
+            $$(ids.dataSource).setValue(oldId || "");
+            $$(ids.dataSource).unblockEvent();
+         }
+
+         // Set settings.datasourceID
+         const dcSettings = this.CurrentDatacollection.toObj() || {};
+         dcSettings.settings = dcSettings.settings || {};
+         dcSettings.settings.datasourceID = datasourceID;
+         this.CurrentDatacollection.fromValues(dcSettings);
+
+         if (!selectedDatasource.isQuery) {
+            // populate link data collection options
+            this.initLinkDatacollectionOptions();
+
+            // populate link fields
+            this.initLinkFieldOptions(
+               this.CurrentDatacollection?.datacollectionLink?.id || null
+            );
+
+            // re-create filter & sort popups
+            this.initPopupEditors();
+
+            // populate filter & sort popups
+            this.populatePopupEditors();
+
+            this.populateBadgeNumber();
+
+            // show options
+            $$(ids.filterPanel).show();
+            $$(ids.sortPanel).show();
+         } else {
+            // hide options
+            $$(ids.filterPanel).hide();
+            $$(ids.sortPanel).hide();
+         }
+      }
+
+      showFilterPopup($button) {
+         this.filter_popup.show($button, null, { pos: "top" });
+      }
+
+      showSortPopup($button) {
+         this.PopupSortFieldComponent.show($button, null, {
+            pos: "top",
+         });
+      }
+
+      onFilterChange() {
+         const datacollection = this.CurrentDatacollection;
+         if (!datacollection) return;
+
+         const filterValues = this.FilterComponent.getValue();
```

Comment:
If we change `this.FilterComponent` to use the `filtercomplexNew()` like I mentioned earlier, that now has a `.isConditionComplete()` method that does this rule check itself.